### PR TITLE
feat: Sticky search pagination

### PR DESCRIPTION
**Pull request by tlgimenes**
## What's the purpose of this pull request?
This PR adds sticky search pagination. 

The idea is that our search uses infinite scroll and, every time the user scrolls down, we need to keep the state of the pagination so once the user navigates on the site and goes back to a search page, he's exactly where he's left.

The before/after this feature can be seen below:
<div style="display: flex; justify-content: space-around; width: 100%">

 
</div>

### CODE REVIEW

File: cypress/support/commands.js [hunk 1]
```diff
@@ -32,8 +32,12 @@ Cypress.Commands.add('waitForHydration', () => {
   return cy.get(`[data-testid=react-hydrated]`).should('exist')
 })
 
-Cypress.Commands.add('clearIDB', () => {
-  return indexedDB.deleteDatabase('keyval-store')
+Cypress.Commands.add('clearStorage', () => {
+  cy.window()
+    .then((win) => {
+      win.sessionStorage.clear()
+    })
```
Comment (victorhmp) [inline cypress/support/commands.js]:
```suggestion
    .then((window) => {
      window.sessionStorage.clear()
    })
```

File: cypress/integration/plp.test.js [hunk 2]
```diff
@@ -75,3 +75,101 @@ describe('Search page Filters and Sorting options', () => {
...
+
+  it('Sticks to last seen page on plp pagination', () => {
+    cy.visit(pages.collection, options)
+    cy.waitForHydration()
+
+    cy.getById('product-link')
+      .should('exist')
+      .should('have.length.gt', 0)
+      .then(($links) => {
+        const before = $links.length // number of products before showMore is clicked
```
Comment (victorhmp) [inline cypress/integration/plp.test.js:117]:
```suggestion
        // Number of products before showMore is clicked
        const before = $links.length
```

File: cypress/integration/plp.test.js [hunk 3]
```diff
@@ -75,3 +75,101 @@ describe('Search page Filters and Sorting options', () => {
...
+
+        cy.getById('show-more')
+          .should('exist')
+          .click()
+          .then(() => {
+            let skuIdBeforeNavigate // The skuId of the last product on the page
```
Comment (victorhmp) [inline cypress/integration/plp.test.js:123]:
```suggestion
            // The skuId of the last product on the page
            let skuIdBeforeNavigate 
```

File: cypress/integration/plp.test.js [hunk 4]
```diff
@@ -75,3 +75,101 @@ describe('Search page Filters and Sorting options', () => {
...
+
+            cy.getById('product-link')
+              .should('have.length.gte', before) // number of products after showMore is clicked should be higher
```
Comment (victorhmp) [inline cypress/integration/plp.test.js:126]:
```suggestion
              // Number of products after showMore is clicked should be higher
              .should('have.length.gte', before) 
```

File: src/sdk/search/Provider.tsx [hunk 5]
```diff
@@ -1,12 +1,12 @@
-/* eslint-disable @typescript-eslint/restrict-plus-operands */
-import React, { createContext, useMemo, useState } from 'react'
+/* eslint-disable react/prop-types */
```
Comment (victorhmp) [inline src/sdk/search/Provider.tsx:1]: Why is this necessary? 🤔

File: src/sdk/search/Provider.tsx [hunk 6]
```diff
@@ -71,7 +79,7 @@ export const SearchProvider: FC<Props> = ({
   children,
   pageInfo,
 }) => {
-  const [pages, setPages] = useState([initalState.page])
+  const [pages, setPages] = useState(() => [initalState.page])
```
Comment (victorhmp) [inline src/sdk/search/Provider.tsx:82]: Can we avoid having two functions with names `setPage` and `setPages`? They're being used in close proximity and inside the callback for `useMemo` they're being used in the same scope.

How about calling the innermost one (inside the callback) `setCurrentPage`, or something like that? Just to make them easier to differentiate and improve readability.

**Review by victorhmp — CHANGES_REQUESTED** [review]

File: src/sdk/search/Provider.tsx (hunk 5 above)
Comment (tlgimenes) [reply src/sdk/search/Provider.tsx:1]: I have no idea. My eslint is complaining about it

File: src/sdk/search/Provider.tsx (hunk 6 above)
Comment (tlgimenes) [reply src/sdk/search/Provider.tsx:82]: what about setPage -> pushPage?

File: src/sdk/search/Provider.tsx (hunk 5 above)
Comment (victorhmp) [reply src/sdk/search/Provider.tsx:1]: That's weird, it might be something wrong in our ESLint setup then. The docs for this specific rule is this one: https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/prop-types.md.

Since we're using TypeScript, and typing the `SearchProvider` props, this error should not be happening

File: src/sdk/search/Provider.tsx (hunk 6 above)
Comment (victorhmp) [reply src/sdk/search/Provider.tsx:82]: Yep, sounds good 😄

File: src/sdk/search/Provider.tsx (hunk 5 above)
Comment (tlgimenes) [reply src/sdk/search/Provider.tsx:1]: nice. I'll remove it then

Comment (victorhmp) [reply src/sdk/search/Provider.tsx:1]: I guess this could stay here! I would just add a comment explaining why we need it

**Review by victorhmp — APPROVED** [review]
